litestar/response/file.py

```diff
@@ -125,8 +125,14 @@ def __init__(
                 providing an :class:`os.stat_result`.
         """
         if not media_type:
-            mimetype, _ = guess_type(filename) if filename else (None, None)
+            mimetype, _encoding = guess_type(filename) if filename else (None, None)
             media_type = mimetype or "application/octet-stream"
+            if _encoding is not None:
+                encoding_header = {"content-encoding": _encoding}
+                if headers is None:
+                    headers = encoding_header
+                else:
+                    headers.update(encoding_header)
```

Comment:
```suggestion
            if encoding is not None:
                headers = headers or {}
                headers.update({"content-encoding": encoding})
```
This could also be something like this?

Reply:
Sure, using this pattern with a different variable name.